### PR TITLE
on backend registered, reconnect to terminals

### LINKED ISSUE #159639: Task reconnection status does not work for default build task
Recording 2022-08-31 at 08 20 47

I urge to selfhost on this for a bit. Status bar never shows the state for me, I think never has after window reload.

### PR DESCRIPTION
**Pull request by meganrogge**
fixes https://github.com/microsoft/vscode/issues/159639

Such that the view and its state does not determine when terminals are reconnected to so tasks get reconnected ASAP

### CODE REVIEW

File: src/vs/workbench/contrib/terminal/browser/terminalService.ts [hunk 1]
```diff
@@ -262,47 +260,45 @@ export class TerminalService implements ITerminalService {
 	}
 
 	handleNewRegisteredBackend(backend: ITerminalBackend) {
```
Comment (meganrogge) [inline src/vs/workbench/contrib/terminal/browser/terminalService.ts:262]: was there a reason we were only doing this for remote terminals before @Tyriar ?

Comment (meganrogge) [reply src/vs/workbench/contrib/terminal/browser/terminalService.ts:262]: don't think so as the local terminal promise was within the block

File: src/vs/workbench/contrib/terminal/browser/terminalService.ts [hunk 2]
```diff
@@ -262,47 +260,45 @@ export class TerminalService implements ITerminalService {
...
-		if (backend.remoteAuthority === this._environmentService.remoteAuthority) {
-			this._primaryBackend = backend;
```
Comment (Tyriar) [inline src/vs/workbench/contrib/terminal/browser/terminalService.ts:266]: I'm pretty sure removing this if will introduce some bugs, this sets the primary backend potentially to secondary ones (eg. local when on a remote window) and breaks `_primaryBackendRegistered`.

**Review by Tyriar — CHANGES_REQUESTED** [review]

File: src/vs/workbench/contrib/terminal/browser/terminalService.ts (hunk 2 above)
Comment (meganrogge) [reply src/vs/workbench/contrib/terminal/browser/terminalService.ts:266]: `_primaryBackendRegistered` was only used by the terminal view. i'll make a build to verify that it doesn't break remote, but local works as expected

Comment (meganrogge) [reply src/vs/workbench/contrib/terminal/browser/terminalService.ts:266]: in fact, I should remove `_primaryBackendRegistered` as it's not used anymore

Comment (Tyriar) [reply src/vs/workbench/contrib/terminal/browser/terminalService.ts:266]: `_primaryBackend` is used in a lot of places though

Comment (meganrogge) [reply src/vs/workbench/contrib/terminal/browser/terminalService.ts:266]: right, didn't remove that part just moved it -

https://github.com/microsoft/vscode/pull/162953/files#diff-f1883fa148c75292499e7150b63a75610d76d58bb20cefff0213d1f5ec6b88fcR258

Comment (Tyriar) [reply src/vs/workbench/contrib/terminal/browser/terminalService.ts:266]: We shouldn't be relying on the order in which they're registered, maybe just wrapping it in an if is enough, I'm not sure though. This stuff is a little fragile

```ts
if (backend.remoteAuthority === this._environmentService.remoteAuthority) {
	this._primaryBackend = backend;
}
```

Comment (meganrogge) [reply src/vs/workbench/contrib/terminal/browser/terminalService.ts:266]: tested on WSL and still works. yeah that seems good to add

**Comment by meganrogge** [comment]
maybe this approach is wrong. in wsl, i'm seeing two terminals get created on reload that pt to the same xterm instance

**Comment by meganrogge** [comment]
I had the wrong approach here

**Pull request closed without merging**